management/WFCORE-1110_Custom_HTTP_Management_Headers.adoc

```diff
@@ -0,0 +1,139 @@
+= WFCORE-1110 Custom HTTP Headers for the HTTP Management Interface
+:author:            Darran Lofthouse
+:email:             darran.lofthouse@redhat.com
+:toc:               left
+:icons:             font
+:idprefix:
+:idseparator:       -
+
+== Overview
+
+The HTTP management endpoint of the application server already returns a pre-defined set of HTTP headers in all of the responses that are sent to the clients, however it is sometimes desirable that additional headers can be returned to satisfy the requirements of an environment the application server is installed within.  This enhancement is to add support for an end user to specify a custom set of HTTP headers that should be returned within each response.
+
+== Issue Metadata
+
+=== Issue
+
+* https://issues.jboss.org/browse/WFCORE-1110[WFCORE-1110]
+
+=== Related Issues
+
+* https://issues.jboss.org/browse/WFLY-12787[WFLY-12787]
+* https://issues.jboss.org/browse/EAP7-1335[EAP7-1335]
+
+=== Dev Contacts
+
+* mailto:{email}[{author}]
+
+=== QE Contacts
+
+* mailto:tterem@redhat.com[Tomas Terem]
+
+=== Testing By
+// Put an x in the relevant field to indicate if testing will be done by Engineering or QE. 
+// Discuss with QE during the Kickoff state to decide this
+[ ] Engineering
+
+[X] QE
+
+=== Affected Projects or Components
+
+=== Other Interested Projects
+
+== Requirements
+
+=== Hard Requirements
+
+The HTTP management interface provides access to a number of contexts, our primary contexts are accessed at the following paths: -
+
+ * `/management` - The endpoint for servicing management requests over HTTP.
+ * `/console` - The endpoint to provide the admin console initial page and complete JavaScript.
+ 
+Subsystems can also dynmically register additional contexts to dynamically provide additional functionality or content over the HTTP management interface. 
+
+As we have different endpoints it may be desirable for them to return different headers, as an example some headers are only applicable to the handling of pages or interpretation of the JavaScript so should only be applied to the `/console` endpoint whilst other headers may be applicable to both.
+
+Configuration will be added to the HTTP management endpoint to allow headers to be specified, the following will be apply when applying the headers: -
+
+ * The configuration will allow an end user to specify headers to be added based on a specified path prefix.
```

Comment:
We should add [this clarification ](https://github.com/wildfly/wildfly/pull/12809/files#diff-f724c56d2b4f9096cfe9b9739d0c889cR98)here as well.

Reply:
For the analysis we need to focus on communicating the initial intent, the whole point of these documents was to share our plans - if we are now cross referencing the community documentation as a recommendation for a change then everyone on the feature change already has access to that information as it is in the documentation.

Reply:
It's good that it's in the documentation, but I believe that the analysis should describe new feature in a way that is clear so there was no misunderstandings later. This behaviour will need to be tested and it isn't obvious from the analysis now so that's why I wanted it included.